cfme/markers/env_markers/provider.py

```diff
@@ -296,6 +297,11 @@ def get_valid_providers(provider):
             allowed_providers = [matching_provs[0]]
         else:
             allowed_providers = []
+    elif selector == SECOND:
+        if matching_provs:
+            allowed_providers = [matching_provs[1]]
```

Comment:
Any chance of an index error if the `len` of matching_provs is 1?

Reply:
I think it will cause an error at setup

Reply:
I was curious so I checked after commenting out `azure_msdn` in yamls,
```
cfme/markers/env_markers/provider.py:402: in providers_by_class
    return providers(metafunc, filters=[pf], selector=selector, fixture_name=fixture_name)
cfme/markers/env_markers/provider.py:302: in providers
    allowed_providers = [matching_provs[1]]
E   IndexError: list index out of range
```
I think it would be helpful caught this error and gave the user some indication that setup failed because there was only 1 provider available. What do you think?